### PR TITLE
algorithm: LCA by binary lifting

### PR DESCRIPTION
**Pull request by Adrito-M**
[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### Describe your change:

- [x] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### Checklist:

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new JavaScript files are placed inside an existing directory.
- [x] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
- [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/JavaScript/pull/1237"></a>

### CODE REVIEW

**Comment by Adrito-M** [comment]
@raklaptudirm @appgurueu  Please Take A Look

**Review by appgurueu — CHANGES_REQUESTED** [review]
This seems to be duplicating the binary lifting you added in https://github.com/TheAlgorithms/JavaScript/pull/1218/. Find a way to reuse the code (which may involve modifying it) rather than duplicating it.

File: Graphs/LCABinaryLifting.js [hunk 1]
```diff
@@ -6,47 +6,21 @@
...
  */
 
+import { BinaryLifting } from './BinaryLifting'
+
 class LCABinaryLifting {
   constructor (root, tree) {
-    this.root = root
-    this.connections = new Map()
-    this.up = new Map() // up[node][i] stores the 2^i-th  parent of node
+    this.binaryLifting = new BinaryLifting(root, tree)
```
Comment (appgurueu) [inline Graphs/LCABinaryLifting.js:13]: This is rather dirty. Why not use inheritance here?

Comment (Adrito-M) [reply Graphs/LCABinaryLifting.js]: Made use of inheritance in code

**Review by appgurueu — APPROVED** [review]

**Review by raklaptudirm — APPROVED** [review]

**Pull request merged**